src/test/java/edu/cornell/kfs/coa/fixture/A21SubAccountFixture.java

```diff
@@ -0,0 +1,65 @@
+package edu.cornell.kfs.coa.fixture;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.kuali.kfs.coa.businessobject.A21IndirectCostRecoveryAccount;
+import org.kuali.kfs.coa.businessobject.A21SubAccount;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+public enum A21SubAccountFixture {
+	A21_SUB_ACCOUNT_1111111_2222222_98_2(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_2222222_2_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(), null),
+
+	A21_SUB_ACCOUNT_1111111_98_INACTIVE_98_INACTIVE_3333333_100_PERCENT_ACTIVE(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange()),
+
+	A21_SUB_ACCOUNT_3333333_100(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(), null, null),
+
+	A21_SUB_ACCOUNT_3333333_100_INACTIVATE(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_2222222_2_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange());
+
+	public A21IndirectCostRecoveryAccount icrAccount1;
+	public A21IndirectCostRecoveryAccount icrAccount2;
+	public A21IndirectCostRecoveryAccount icrAccount3;
+
+	private A21SubAccountFixture(A21IndirectCostRecoveryAccount icrAccount1,
+			A21IndirectCostRecoveryAccount icrAccount2,
+			A21IndirectCostRecoveryAccount icrAccount3) {
```

Comment:
I would suggest changing this fixture to instead use an array of ICR accounts, and to update the constructor to use a var-args parameter instead of three individual ones. This will greatly simplify the setup process.